new-version.sh

```diff
@@ -0,0 +1,172 @@
+
+# Change working directory to project root
+project_root=$(git rev-parse --show-toplevel)
+cd "${project_root}"
+
+# Verify bump2version is installed
+if [[ ! $(type -P bump2version) ]]; then
+  echo "bump2version not installed! Please see https://github.com/c4urself/bump2version#installation"
+  exit 1;
+fi
+
+# Verify redoc-cli is installed
+if [[ ! $(type -P redoc-cli) ]]; then
+  echo "redoc-cli not installed! Please see https://www.npmjs.com/package/redoc-cli"
+  exit 1;
+fi
+
+branch=$(git symbolic-ref --short HEAD)
+if [[ "${branch}" != "main" ]]; then
+  echo "error: you may only release on 'main'!"
+  exit 1;
+fi
+
+if [[ $# -eq 0 ]] ; then
+  usage
+fi
+
+# Ensure no unstaged changes are present in working directory
+if [[ -n "$(git status --porcelain --untracked-files=no)" ]] ; then
+  echo "error: you have unstaged changes in your working directory!"
+  exit 1;
+fi
+
+while [ $# -gt 0 ]; do
+  case $1 in
+    '--release-version'|-r)
+       shift
+       RELEASE_VERSION="${1}"
+       ;;
+    '--next-version'|-n)
+       shift
+       NEXT_VERSION="${1}"
+       ;;
+    '--push'|-p)
+       shift
+       PUSH="${1}"
+       ;;
+    '--help'|-h)
+       usage
+       ;;
+    *) exit 1
+       ;;
+  esac
+  shift
+done
+
+# Append '-SNAPSHOT' to 'NEXT_VERSION' if not a release candidate, or missing
+if [[ ! "${NEXT_VERSION}" == *-rc.? &&
+      ! "${NEXT_VERSION}" == *-SNAPSHOT ]]; then
+  NEXT_VERSION="${NEXT_VERSION}-SNAPSHOT"
+fi
+
+# Ensure valid versions
+VERSIONS=($RELEASE_VERSION $NEXT_VERSION)
+for VERSION in "${VERSIONS[@]}"; do
+  if [[ ! "${VERSION}" =~ ${SEMVER_REGEX} ]]; then
+    echo "Error: Version '${VERSION}' must match '${SEMVER_REGEX}'"
+    exit 1
+  fi
+done
+
+# Ensure python module version matches X.Y.Z or X.Y.ZrcN (see: https://www.python.org/dev/peps/pep-0440/),
+PYTHON_RELEASE_VERSION=${RELEASE_VERSION}
+if [[ "${RELEASE_VERSION}" == *-rc.? ]]; then
+  RELEASE_CANDIDATE=${RELEASE_VERSION##*-}
+  PYTHON_RELEASE_VERSION="${RELEASE_VERSION%-*}${RELEASE_CANDIDATE//.}"
+fi
+
+# (1) Bump python module versions
+PYTHON_MODULES=(client/python/ integration/common/ integration/airflow/)
+for PYTHON_MODULE in "${PYTHON_MODULES[@]}"; do
+  (cd "${PYTHON_MODULE}" && bump2version manual --new-version "${PYTHON_RELEASE_VERSION}" --allow-dirty)
+done
+
+# (2) Bump java module versions
+sed -i  "s/^version=.*/version=${RELEASE_VERSION}/g" ./integration/spark/gradle.properties
+sed -i  "s/^version=.*/version=${RELEASE_VERSION}/g" ./client/java/gradle.properties
+
+# (4) Bump version in docs
+sed -i  "s/^  version:.*/  version: ${RELEASE_VERSION}/g" ./spec/OpenLineage.yml
+sed -i  "s/<version>.*/<version>${RELEASE_VERSION}<\/version>/g" ./integration/spark/README.md
+sed -i  "s/openlineage-spark:.*/openlineage-spark:${RELEASE_VERSION}/g" ./integration/spark/README.md
+
+# (5) Bundle openAPI docs
+redoc-cli bundle spec/OpenLineage.yml -o docs/OpenLineage.html  --title "OpenLineage API Reference"
```

Comment:
We'll be versioning the OpenLineage spec using [`schemaver`](https://snowplowanalytics.com/blog/2014/05/13/introducing-schemaver-for-semantic-versioning-of-schemas) and [`semver`](https://semver.org/) for our clients. Mind removing lines [L144](https://github.com/OpenLineage/OpenLineage/blob/main/new-version.sh#L144) and [L149](https://github.com/OpenLineage/OpenLineage/blob/main/new-version.sh#L149) from the release script?